pixie/stdlib.pxi

```diff
@@ -1304,9 +1312,13 @@ and implements IAssociative, ILookup and IObject."
                                           fields)))
                         `(-hash [self]
                                 (hash [~@field-syms]))]
-        deftype-decl `(deftype ~nm ~fields ~@default-bodies ~@body)]
+        deftype-decl `(deftype ~nm ~(conj fields '__meta) ~@default-bodies ~@body)
+        ctor `(defn ~ctor-name ~field-syms
```

Comment:
The constructor from the deftype includes the __meta field, so we need to "overwrite" it to allow construction of the record with __meta initalized to nil (or we get a wrong number of arg error).